core/src/main/java/org/axonframework/eventhandling/saga/LoggingSagaErrorHandler.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2010-2017. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.axonframework.eventhandling.saga;
+
+import org.axonframework.eventhandling.EventMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.String.format;
+
+/**
+ * Implementation of {@link SagaInvocationErrorHandler} that logs exceptions as errors but otherwise does nothing to
+ * prevent event handling from continuing.
+ *
+ * @author Milan Savic
+ * @since 3.2
+ */
+public class LoggingSagaErrorHandler implements SagaInvocationErrorHandler {
```

Comment:
Can't this also be solved by letting the `LoggingErrorHandler` implement the `SagaInvocationErrorHandler`, like you did for the `PropagatingErrorHandler`?

Reply:
If we go this route, we should probably remove https://github.com/AxonFramework/AxonFramework/blob/master/core/src/main/java/org/axonframework/queryhandling/LoggingQueryInvocationErrorHandler.java also. What do you think?

Reply:
Hmm not that weird an assumption, lets see what @abuijze thinks about that.

Reply:
Also, one thing to consider is package of `LoggingErrorHandler`, because we (probably) don't want to have dependencies between `eventhandling` and `queryhandling` packages if we go with single `LoggingErrorHandler`.

Reply:
While I like the idea to restructure the code, we can only do this in a major release. Between releases, the APIs (at least those of main/public classes)must remain stable.
Also, for Commands and Queries, feedback can be given to the component that sent out the Message. For Events, this is not possible. Therefore, having a (good) error handling mechanism for Event Handlers is crucial.

I was actually thinking that we probably don't need the `SagaInvocationErrorHandler` interface. The `EventListener` parameter is provided to allow an ErrorHandler to re-invoke the listener if the exception is a transient one. This could be exactly the same for the Saga instance.

As a bit of background: I like to think of a Saga as a stateful event handler, whose creation and state changes are triggered by events. In that sense, they shouldn't be treated very differently.

Reply:
Interesting idea, we could remove `SagaInvocationErrorHandler` and use `ListenerInvocationErrorHandler` instead. Now, since in `AbstractSagaManager` we are dealing with `Saga` and not `EventListener` we could pass method reference to `Saga#handle(EventMessage)` as `eventListener` parameter. It would look like something like this: `listenerInvocationErrorHandler.onError(exception, event, saga::handle);`. However, this approach limits implementation of error handler because the error handler does not have access to `Saga`.

Similar but a bit cleaner approach would be to make `Saga` interface extend `EventListener` interface which would avoid method referencing. I guess that if we go this way, we would have to change the API a lot.

Last option would be to leave `SagaInvocationErrorHandler` as is.

Opinions?

Reply:
Hi @m1l4n54v1c,

I really like the line of thinking to have `Saga` implement `EventListener`. It fits well with the thought that a `Saga` is essentially a stateful `EventListener`.
The difficulty is that Saga currently defines a method whose signature conflicts with the `handle` method of `EventListener`. This would be a breaking API change. Since this feature is scheduled for 3.2, and I don't consider this API to be 'public', it would be fine to do that.

Instead of `boolean handle(EventMessage)`, it would have `boolean canHandle(...)` and `void handle(...)`. `canHandle` would need to check if the Saga has the required associations (and 'active' state) to handle the incoming event.

This would make the additional `SagaInvocationErrorHandler` superfluous.

Reply:
Cool, will update the PR accordingly. `QueryInvocationErrorHandler` will be left as is.